tests/contrib/pytest/test_pytest.py

```diff
@@ -25,6 +27,33 @@
 from tests.utils import override_env
 
 
+def _get_spans_from_list(
+    spans: t.List[ddtrace.Span], span_type: str, name: t.Optional[str] = None, status: t.Optional[str] = None
+) -> t.List[ddtrace.Span]:
+    _names_map = {
+        "session": ("test_session_end",),
+        "module": ("test_module_end", "test.module"),
+        "suite": ("test_suite_end", "test.suite"),
+        "test": ("test", "test.name"),
+    }
+
+    if span_type == "session" and name is not None:
+        raise ValueError("Cannot get session spans with a name")
+
+    target_type = _names_map[span_type][0]
+    target_name = _names_map[span_type][1] if name else None
+    return [
+        span
+        for span in spans
+        if span.get_tag("type") == target_type
+        and (
+            span.get_tag(target_name) == name
+            if name
+            else True and (span.get_tag("test.status") == status if status else True)
```

Comment:
Is there a way to write this condition that makes it a little clearer what it's trying to do?

Reply:
Yeah, this thing's getting a little unwieldy for a comprehension.

I'll rewrite it.

Reply:
Also I think this gets parsed as `else (True and ...)`.

Reply:
https://github.com/DataDog/dd-trace-py/pull/10180/files#diff-48a91ef11a9dbcd767fd3050fa39ca48a9bc8454d47d71b2afc35ebc630686f7R48-R59

I don't love it but.. at least it's more readable.